src/PhpSpec/Matcher/ThrowMatcher.php

```diff
@@ -121,11 +121,25 @@ public function verifyPositive($callable, array $arguments, $exception = null)
         }
 
         if (!$exceptionThrown instanceof $exception) {
+            $format = 'Expected exception of class %s, but got %s.';
+
+            // Show full error message if the thrown exception is an instance of Error.
+            // See https://github.com/phpspec/phpspec/issues/908
+            if ($exceptionThrown instanceof \Error) {
+                // TODO: Can we can use $this->presenter->presentValue() rather
+                // than adding the raw Error message in quotes here? Currently
+                // not included since presentValue() truncates the string to 25
+                // characters but we want to show the full Error message for
+                // debugging purposes.
+                $format = 'Expected exception of class %s, but got %s with the message: "%s"';
+            }
+
             throw new FailureException(
                 sprintf(
-                    'Expected exception of class %s, but got %s.',
+                    $format,
                     $this->presenter->presentValue($exception),
-                    $this->presenter->presentValue($exceptionThrown)
+                    $this->presenter->presentValue($exceptionThrown),
+                    $exceptionThrown->getMessage()
```

Comment:
I'm not sure how I feel about this silently-ignored parameter... is there a way to make the conditional more explicit?